### PR TITLE
perf(apps): Only load `PluginSourceFile`s instead of `Plugin.archive`

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

Related to #10308.
Optimizes loading of plugins in the plugin server using the Django-side `Plugin.archive` extraction mechanism introduced in #10333. Details in the original PR.

## How did you test this code?

Contains plugin server tests.

### CODE REVIEW

File: plugin-server/src/types.ts [hunk 1]
```diff
@@ -288,6 +289,8 @@ export interface Plugin {
     is_stateless?: boolean
     public_jobs?: Record<string, PluginPublicJobPayload>
     log_level?: PluginLogLevel
+    latest_tag?: string
```
Comment (macobo) [inline plugin-server/src/types.ts:292]: What are these? Do we require them in the plugin server?

File: plugin-server/src/utils/db/sql.ts [hunk 2]
```diff
@@ -25,11 +25,38 @@ function pluginConfigsInForceQuery(specificField?: keyof PluginConfig): string {
 
 export async function getPluginRows(hub: Hub): Promise<Plugin[]> {
     const { rows }: { rows: Plugin[] } = await hub.db.postgresQuery(
+        // `posthog_plugin` columns have to be listed individually, as we want to exclude the blob `archive` column,
+        // and Postgres unfortunately doesn't have a feature to exclude a single column from *
         `SELECT
```
Comment (macobo) [inline plugin-server/src/utils/db/sql.ts:30]: Q: This query took ~13 seconds before individually and we're adding to this with a new join here - not sure how this is resolving the performance problem outlined in https://github.com/PostHog/posthog/issues/10308 which was focussed on the query only.

File: plugin-server/src/types.ts (hunk 1 above)
Comment (Twixes) [reply plugin-server/src/types.ts:292]: I don't think we do, but this is returned and I'm strongly in favor of types being fully representative of the data structures. Though I can also just explicitly omit this in `getPluginRows()`. What do you think?

Comment (macobo) [reply plugin-server/src/types.ts:292]: Let's explicitly not fetch data not being used by plugin server.

File: plugin-server/src/utils/db/sql.ts (hunk 2 above)
Comment (Twixes) [reply plugin-server/src/utils/db/sql.ts:30]: The query itself, as in SQL, is not an issue. It's reasonably well optimized with regard to lookups (incl. joins) and there's just not many rows. What scales badly is the `archive` column. It takes time to transfer all this and unpack it in the plugin server each time. Locally it's 10-30 s with the `archive` column and 0.3-0.5 s when excluding it. In the production Django shell it's 2-3 s vs. 0.3-0.5 s. (This does not include unpacking.)

TBH only now did I notice that there's one big offender which is the main source of our current woes – turns out that https://github.com/PostHog/posthog-variance-plugin includes the `.yarn` folder, which is 40 MB in size. All the other archives are 10 MB _in total_, so optimizing the Variance plugin could go a long way.

However, I think it's still worth it to go through this refactor, as with the optimized scheme this PR introduces we will hopefully not have to worry about archive sizes in the plugin server for a long time. We've now also got metrics to track this.

Comment (macobo) [reply plugin-server/src/utils/db/sql.ts:33]: It is the most time-consuming query we do on postgres (total time spent): https://metabase.posthog.net/question/344-most-time-consuming-postgres-queries. This means that during deploys we risk bringing down capture and decide endpoints and causing other load-related issues.

Can you run and compare the new and old queries - will we see a performance win from removing the `archive` column?

> However, I think it's still worth it to go through this refactor, as with the optimized scheme this PR introduces we will hopefully not have to worry about archive sizes in the plugin server for a long time. We've now also got metrics to track this.

Not arguing that - I think this is great, but orthogonal to the goal in https://github.com/PostHog/posthog/issues/10308 if it doesn't reduce the time spent in postgresql.

Comment (macobo) [reply plugin-server/src/utils/db/sql.ts:33]: I think I won't get the metrics I'm looking for here. Updating the PR description to remove the "closes" note as we'll need to monitor and IMO likely need to make file loading lazy in the future.

Comment (Twixes) [reply plugin-server/src/utils/db/sql.ts:33]: Very well, I agree it's not a slam dunk, let's monitor.

**Review by yakkomajuri — APPROVED** [review]
lgtm

**Pull request merged**